packages/http/src/validator/validators/body.ts

```diff
@@ -70,12 +96,12 @@ export function findContentByMediaTypeOrFirst(specs: IMediaTypeContent[], mediaT
   );
 }
 
-function deserializeAndValidate(content: IMediaTypeContent, schema: JSONSchema, target: string, prefix?: string, bundle?: unknown) {
+function deserializeAndValidate(content: IMediaTypeContent, schema: JSONSchema, target: string, prefix?: string, multipartBoundary?: string, bundle?: unknown) {
   const encodings = get(content, 'encodings', []);
-  const encodedUriParams = splitUriParams(target);
 
   return pipe(
-    validateAgainstReservedCharacters(encodedUriParams, encodings, prefix),
+    content.mediaType === "multipart/form-data" ? parseMultipartFormDataParams(target, multipartBoundary) : splitUriParams(target),
```

Comment:
Since parseMultipartFormDataParams returns E.left if the boundary string is empty, the return type is Either, and I wrapped the result of splitUriParams in the Either monad as well to match the return type of parseMultipartFormDataParams. Thus, we move this statement into the expression pipeline.